packages/std/src/math/decimal.rs

```diff
@@ -55,6 +56,30 @@ impl Decimal {
     pub fn is_zero(&self) -> bool {
         self.0 == 0
     }
+
+    /// Returns the approximate square root as a Decimal.
+    ///
+    /// This should not overflow or panic.
+    pub fn sqrt(&self) -> Self {
+        // We start with the highest precision possible and lower it until
+        // there's no overflow.
+        (0..=9)
+            .rev()
+            .find_map(|i| self.sqrt_with_precision(i))
+            .unwrap()
```

Comment:
```suggestion
            // The last step (i = 0) is guaranteed to succeed because `isqrt(u128::MAX) * 10^9` does not overflow
            .unwrap()
```